platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/migration/ConfigurationMigrationManagerTest.java

```diff
@@ -618,14 +618,275 @@ public void testDoImportRecordsMigrationExceptionWithInvalidChecksum() throws Ex
                 + ".dar]."));
   }
 
-  private void expectImportDelegationIsSuccessful()
-      throws NoSuchPaddingException, NoSuchAlgorithmException {
+  @Test
+  public void doDecryptSucceeds() throws Exception {
+    final Path path = ddfHome.resolve(TEST_DIRECTORY);
+
+    configurationMigrationManager = spy(getConfigurationMigrationManager());
+
+    // Need to actually generate a zip with a valid checksum so calling export
```

Comment:
❓ Instead of creating a real zip just to get the checksum, couldn't we mock out the `MigrationZipFile` instead and return `true`/`false` when `isValidChecksum()` is called? This would probably be easier if the decryption code in `DecryptMigrationManagerImpl` was moved to `MigrationZipFile`.

Reply:
Good idea. I also changed all existing import test cases to do the same.